src/contexts/SwapContext.tsx

```diff
@@ -403,6 +358,177 @@ export const SwapContextProvider: FC<{
+    });
+  };
+
+  const getAddressLookupTableAccounts = useCallback(
+    async (keys: string[]): Promise<AddressLookupTableAccount[]> => {
+      const addressLookupTableAccountInfos = await connection.getMultipleAccountsInfo(
+        keys.map((key) => new PublicKey(key)),
+      );
+
+      return addressLookupTableAccountInfos.reduce((acc, accountInfo, index) => {
+        const addressLookupTableAddress = keys[index];
+        if (accountInfo) {
+          const addressLookupTableAccount = new AddressLookupTableAccount({
+            key: new PublicKey(addressLookupTableAddress),
+            state: AddressLookupTableAccount.deserialize(accountInfo.data),
+          });
+          acc.push(addressLookupTableAccount);
+        }
+
+        return acc;
+      }, new Array<AddressLookupTableAccount>());
+    },
+    [connection],
+  );
+
+  const onSubmit = useCallback(async () => {
+    if (!walletPublicKey || !wallet?.adapter || !quoteResponseMeta) {
+      throw new Error('Missing wallet or quote');
+    }
+
+    // Fetch necessary data from APIs
+    const { meta, instructions, onSubmitWithIx } = await onRequestIx();
+    const { blockhash, lastValidBlockHeight } = await connection.getLatestBlockhash();
+
+    // Destructure instructions
+    const {
+      setupInstructions,
+      swapInstruction: swapInstructionPayload,
+      cleanupInstruction,
+      addressLookupTableAddresses,
+      // Note: Not using this as we need to calculate the optimal compute unit limit and price
+      computeBudgetInstructions,
+    } = instructions;
+
+    // Prepare address lookup table accounts
+    const addressLookupTableAccounts = await getAddressLookupTableAccounts(addressLookupTableAddresses);
+
+    // Prepare transaction instructions
+    const setupTransactionInstructions = setupInstructions.map(deserializeInstruction);
+    const swapTransactionInstruction = deserializeInstruction(swapInstructionPayload);
+    const cleanupTransactionInstruction = cleanupInstruction ? deserializeInstruction(cleanupInstruction) : null;
+
+    const filteredInstructions = [
+      ...setupTransactionInstructions,
+      swapTransactionInstruction,
+      cleanupTransactionInstruction,
+    ].filter(Boolean) as TransactionInstruction[];
+
+    // Prepare payer key
+    const payerKey = new PublicKey(walletPublicKey);
+
+    // Set optimal compute unit limit and price
+    let referenceFee = 0;
+    if (referenceFees?.jup) {
+      switch (priorityLevel) {
+        case 'MEDIUM':
+          referenceFee = referenceFees.swapFee;
+          break;
+        case 'HIGH':
+          referenceFee = referenceFees.swapFee * PRIORITY_LEVEL_MULTIPLIER_HIGH;
+          break;
+        case 'VERY_HIGH':
+          referenceFee = referenceFees.swapFee * PRIORITY_LEVEL_MULTIPLIER_VERY_HIGH;
+          break;
+        default:
+          break;
+      }
+    }
+    const { units, microLamports } = await getOptimalComputeUnitLimitAndPrice({
```

Comment:
Have to use the instruction method here to construct the `Transaction` because we want to get the optimal `computeUnitLimit`  by inserting the `ComputeBudgetProgram.setComputeUnitLimit` instruction using `MAX_COMPUTE_UNIT_LIMIT (1_400_000)` to get the real compute units used

See
https://github.com/solana-developers/helpers/blob/a7cfe108c0d49348995090cb7550eec7b983807f/src/index.ts#L336-L342